client/pkg/session/session.go

```diff
@@ -0,0 +1,176 @@
+package session
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"sync/atomic"
+
+	"github.com/wandb/wandb/client/internal/connection"
+	"github.com/wandb/wandb/client/internal/launcher"
+	"github.com/wandb/wandb/core/pkg/observability"
+	"github.com/wandb/wandb/core/pkg/service"
+)
+
+const (
+	localHost = "127.0.0.1"
+)
+
+// SessionParams holds parameters for creating a new Session
+type SessionParams struct {
```

Comment:
can just call this `Params`, so that it's referenced as `session.Params{}` outside the package